app/src/ereferrals/java/org/eyeseetea/malariacare/domain/usecase/LoginUseCase.java

```diff
@@ -148,7 +146,9 @@ private void pullOrganisationCredentials() {
                     true);
         }
 
-        mCredentialsLocalDataSource.saveOrganisationCredentials(orgUnitCredentials);
+        if (orgUnitCredentials != null) {
```

Comment:
if orgUnitCredentials can be null if network fail to retrieve credentials from the server then in the checkUserCredentialsWithOrgUnit method we should validate null